main.go

```diff
@@ -18,6 +18,10 @@ func main() {
 	maxSize := flag.Int64("max_size", -1,
 		"The maximum size of the remote cache in GiB. This flag is required.")
 	htpasswd_file := flag.String("htpasswd_file", "", "Path to a .htpasswd file. This flag is optional. Please read https://httpd.apache.org/docs/2.4/programs/htpasswd.html.")
+	tls_enabled := flag.Bool("tls_enabled", false, "Bool specifying wheather or not to start the server with tls.  If true, default port changes to 8443, and server_cert and server_key flags are requred.")
```

Comment:
I wonder if a flag magically changing the default value of another flag is a good idea, it seems it could confuse people? What do you think?